src/clients/c++/library/grpc_client.cc

```diff
@@ -1046,10 +1048,23 @@ InferenceServerGrpcClient::PreRunProcessing(
         }
       }
     }
+    index++;
   }
 
+  // Remove extra InferInputTensor submessages, that are not required for
+  // this request.
+  if (index < infer_request_.inputs().size()) {
```

Comment:
This needs to be "while" with index++ in the loop. The reason it is more efficient is that the other way may have shifted array elements... depending on how it was implemented.

Reply:
Ahh.. I am making many mistakes today!
About index++, after the RemoveLast call, infer_request_.inputs().size() will be decremented by 1, so we must be fine.